### DISCUSSION TITLE
Understanding Sktime's BaseForecaster API - Why doesn't fit_predict() have separate arguments for X of prediction mode especially for forecasting with exogenous data

### DISCUSSION THREAD

**Discussion by RahulDas-dev** [alan-turing-institute/sktime#4949] [General]
I'm relatively new to Sktime and have been exploring the `BaseForecaster` API. As I was going through the documentation, I noticed that the `fit()` method requires `y`, `X`, and `fh` for training the forecaster, and the `predict()` method needs `X` and `fh` to make predictions with exogenous data. However, when it comes to the `fit_predict()` method, which combines both training and prediction, there are no separate arguments for X of prediction mode.

Kindly let me know where I am wrong here ??

Here is my question from different angel , Why `fit` + `pedict` and `fit_predict` shows difference in Predicted outputs

Kindly find the small example code bellow

### Data Loading

```python
from sktime.datasets import load_macroeconomic
from sktime.forecasting.compose import make_reduction
from sktime.forecasting.model_selection import temporal_train_test_split
from sklearn.linear_model import ElasticNet
from sktime.utils.plotting import plot_series

data = load_macroeconomic()
data.head()

train_xy, test_xy = temporal_train_test_split(data, test_size=5)
train_y = train_xy.pop('unemp')
test_y = test_xy.pop('unemp')
```
### fit+ predict
```python
forecaster1 = make_reduction(estimator=ElasticNet(random_state=42, max_iter=100000),
                            scitype="tabular-regressor",
                            window_length=5,
                            strategy="recursive",
                            pooling="global")

forecaster1.fit(train_y, X=train_xy, fh=[1,2,3,4,5])
print(f'forecaster1 cutoff {forecaster1.cutoff}')
y_predict1 = forecaster1.predict(fh=[1,2,3,4,5], X=test_xy)
print(f'forecaster1 cutoff {forecaster1.cutoff}')
print(y_predict1)
plot_series(train_y.tail(12), test_y,y_predict1, labels=["Train", "Test", "predict"])
```
#### output
```
forecaster1 cutoff PeriodIndex(['2008Q2'], dtype='period[Q-DEC]', name='Period')
forecaster1 cutoff PeriodIndex(['2008Q2'], dtype='period[Q-DEC]', name='Period')
2008Q3     4.793714
2008Q4     6.430364
2009Q1     9.116874
2009Q2    10.521908
2009Q3     9.619637
Freq: Q-DEC, Name: unemp, dtype: float64
```
![image](https://github.com/sktime/sktime/assets/32712591/ff2aa5bd-c43d-4345-8fab-c72af7ab5d78)

### fit_predict
```python
forecaster2 = make_reduction(estimator=ElasticNet(random_state=42, max_iter=100000),
                            scitype="tabular-regressor",
                            window_length=5,
                            strategy="recursive",
                            pooling="global")


y_predict2 = forecaster2.fit_predict(train_y, X=train_xy, fh=[1,2,3,4,5])
print(f'forecaster2 cutoff {forecaster2.cutoff}')
print(y_predict2)
plot_series(train_y.tail(12), test_y,y_predict2, labels=["Train", "Test", "predict"])
```
#### output
```
forecaster2 cutoff PeriodIndex(['2008Q2'], dtype='period[Q-DEC]', name='Period')
2008Q3    11.373905
2008Q4    11.373905
2009Q1    11.373905
2009Q2    11.373905
2009Q3    11.373905
Freq: Q-DEC, Name: unemp, dtype: float64
```
![image](https://github.com/sktime/sktime/assets/32712591/ad0a768c-c7d5-4e8d-acc0-8e99cccf78a2)

### Additional Information

- Sktime version: 0.20.0
- Python version: 3.9.7

**Comment by fkiraly** [comment]
hm, very interesting observation!

Currently, `fit_predict` passes the same `X` to `fit` and `predict`.
That is, `fit_predict(y=y, X=X, fh=fh)` is the same as `fit(y, X=X, fh=fh).predict(X=X)
For some estimators, this can result in a different `fit`.

Do you think there should be two separate `X`-like parameters in `fit_predict`? Or should sth else happen?
